src/input_parsers/OnnxParser.cpp

```diff
@@ -0,0 +1,1281 @@
+    if ( attr == nullptr )
+    {
+        missingAttributeError( node, name );
+    }
+    return attr->t();
+}
+
+Vector<int> getIntsAttribute( onnx::NodeProto& node, String name, Vector<int>& defaultValue )
+{
+    const onnx::AttributeProto* attr = findAttribute( node, name, onnx::AttributeProto_AttributeType_INT );
+    if ( attr == nullptr )
+    {
+        return defaultValue;
+    }
+
+    Vector<int> result;
+    for ( int i = 0; i < attr->ints_size(); i++ )
+    {
+        result.append( attr->ints( i) );
+    }
+    return result;
+ }
+
+/*******************
+ * Private methods *
+ *******************/
+
+/**
+ * @brief Initialises the mapping from node names to nodes and
+ * checks the invariant that there are no duplicate names.
+ */
+void OnnxParser::initializeMaps()
+{
+    Set<std::string> names;
+    for ( const onnx::NodeProto& node : _network.node() )
+    {
+        const std::string name = node.name();
+        if ( names.exists( name) )
+        {
+            String errorMessage = "Nodes in Onnx network must have a unique name";
+            throw MarabouError( MarabouError::ONNX_PARSER_ERROR, errorMessage.ascii() );
+        }
+        else
+        {
+            _nodeMap.insert( name, &node );
+            names.insert( name );
+        }
+    }
+
+    for ( const onnx::TensorProto& initializer : _network.initializer() )
+    {
+        const std::string name = initializer.name();
+        if ( names.exists( name ) )
+        {
+            String errorMessage = "Nodes in Onnx network must have a unique name";
+            throw MarabouError( MarabouError::ONNX_PARSER_ERROR, errorMessage.ascii() );
+        }
+        else
+        {
+            _constantMap.insert( name, &initializer );
+            names.insert( name );
+        }
+    }
+}
+
+void OnnxParser::validateUserInputNames( Set<String>& inputNames )
+{
+    // Validate the provided inputs
+    for ( String inputName : inputNames )
+    {
+        if ( !_nodeMap.exists( inputName ) )
+        {
+            String errorMessage = Stringf( "Input %s not found in graph!", inputName.ascii() );
+            throw MarabouError( MarabouError::ONNX_PARSER_ERROR, errorMessage.ascii() );
+        }
+    }
+}
+
+void OnnxParser::validateUserOutputNames( String &outputName )
+{
+    for ( auto node : _network.node() )
+    {
+        for ( String outputNodeName : node.output() )
+        {
+            if ( outputName == outputNodeName )
+                return;
+        }
+    }
+
+    String errorMessage = Stringf( "Output %s not found in graph!", outputName.ascii() );
+    throw MarabouError(  MarabouError::ONNX_PARSER_ERROR, errorMessage.ascii() );
+}
+
+Set<String> OnnxParser::readInputNames()
+{
+    ASSERT( _network.input().size() >= 1 );
+
+    Set<String> initializerNames;
+    for ( auto &initNode : _network.initializer() )
+    {
+        std::cout << "\ninitialiser: " + initNode.name();
```

Comment:
Either remove the print statements or use the "ONNX_LOG" instead so that these messages can be seen only when debugging?